### ISSUE TITLE
HashRegistrar: bid cancellation logs BidRevealed event, not a separate one

### ISSUE THREAD

**Issue by veox** [ethereum/ens#207]
[`unsealBid()`](https://github.com/ethereum/ens/blob/1ce01f29d601f843ac308a015c327c702d224cc7/contracts/HashRegistrarSimplified.sol#L369) and [`cancelBid()`](https://github.com/ethereum/ens/blob/1ce01f29d601f843ac308a015c327c702d224cc7/contracts/HashRegistrarSimplified.sol#L420) log the same `BidRevealed` event.

Finding out which function resulted in that event, when analysing event logs, can be discerned by checking the [`status` argument](https://github.com/ethereum/ens/blob/1ce01f29d601f843ac308a015c327c702d224cc7/contracts/HashRegistrarSimplified.sol#L121), which is packed in the transaction receipt as the lower 32 bytes of `data`. (Actually, just the lowest byte is fine, since there are 6 possible values throughout the code.)

Personally, I find this somewhat convoluted, but workable.

The issue here is that:

* when event happens in `unsealBid()`, it logs the hash bid for, that message's sender, and the bid value;
* when event happens in `cancelBid()`, it logs the unreavealed seal, the original bidder, and a 0.

In other words, the `BidRevealed` logs two kinds of events, that are semantically different.

Suggesting to introduce a separate `BidCancelled` event in the permanent registrar.

**Comment by Arachnid** [comment]
Agreed, these should be different.
